ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java

```diff
@@ -1,364 +1,402 @@
-            });
-    }
-
-    @Override
-    public void onDestroy() {
-        Log.d(TAG, "onDestroy");
-        paused = true;
-        pendingRequestBanner.set(false);
-        if (vungleNativeAd != null) {
-            vungleNativeAd.finishDisplayingAd();
-            mVungleManager.cleanUpBanner(mPlacementForPlay, vungleNativeAd);
-            vungleNativeAd = null;
-        }
-        adLayout = null;
-    }
-
-    //banner
-    @Override
-    public void onPause() {
-        Log.d(TAG, "onPause");
-        paused = true;
-        updateVisibility();
-    }
-
-    @Override
-    public void onResume() {
-        Log.d(TAG, "onResume");
-        paused = false;
-        updateVisibility();
-    }
-
-    @Override
-    public void requestBannerAd(Context context,
-                                final MediationBannerListener mediationBannerListener,
-                                Bundle serverParameters,
-                                AdSize adSize,
-                                MediationAdRequest mediationAdRequest,
-                                Bundle mediationExtras) {
-        Log.d(TAG, "requestBannerAd");
-        pendingRequestBanner.set(true);
-        mMediationBannerListener = mediationBannerListener;
-        AdapterParametersParser.Config config;
-        try {
-            config = AdapterParametersParser.parse(mediationExtras, serverParameters);
-        } catch (IllegalArgumentException e) {
-            Log.w(TAG, "Failed to load ad from Vungle", e);
-            if (mediationBannerListener != null) {
-                mediationBannerListener
-                        .onAdFailedToLoad(VungleInterstitialAdapter.this,
-                                AdRequest.ERROR_CODE_INVALID_REQUEST);
-            }
-            return;
-        }
-
-        mVungleManager = VungleManager.getInstance();
-        mAdConfig = VungleExtrasBuilder.adConfigWithNetworkExtras(mediationExtras);
-        mPlacementForPlay = mVungleManager.findPlacement(mediationExtras, serverParameters);
-
-        if (TextUtils.isEmpty(mPlacementForPlay)) {
-            String message = "Failed to load ad from Vungle: Missing or Invalid Placement ID.";
-            Log.w(TAG, message);
-            mMediationBannerListener
-                    .onAdFailedToLoad(VungleInterstitialAdapter.this,
-                            AdRequest.ERROR_CODE_INVALID_REQUEST);
-            return;
-        }
-        if (!hasBannerSizeAd(adSize)) {
-            String message = "Failed to load ad from Vungle: Invalid banner size.";
-            Log.w(TAG, message);
-            mMediationBannerListener
-                    .onAdFailedToLoad(VungleInterstitialAdapter.this,
-                            AdRequest.ERROR_CODE_INVALID_REQUEST);
-            return;
-        }
-
-        //workaround for missing onPause/onResume/onDestroy
-        adLayout = new RelativeLayout(context) {
-            @Override
-            protected void onWindowVisibilityChanged(int visibility) {
```

Comment:
@HuFangshuai @alexander-gubov This block of code is missing from our current code, any idea if this is incidental or we intentionally removed this, if so why dont we need it any longer

Reply:
I've added it before repo migration due to bug of missing callbacks form AdMob, but it was removed by @HuFangshuai in https://github.com/Vungle/googleads-mobile-android-mediation/pull/17#discussion_r342449222